### DISCUSSION TITLE
Collisions between static and kinematic bodies, integrating logic and physics in a game

### DISCUSSION THREAD

**Discussion by Franckapik** [pmndrs/use-cannon#71] [General] [answered]
I would like to understand why the collision between a mesh cube and the ground works only on position [0,0,0] of the ground ... Is this possible to make body visible on the view to see if i have to scale the shape ?

Mesh are imported from blender and convert to jsx with gtlftojsx.

Thank you a lot !

https://codesandbox.io/s/github/Franckapik/NandY/tree/master/views?file=/src/3d/Game.js

https://github.com/Franckapik/NandY/blob/master/views/src/3d/Game.js

**Comment by stockhuman** [comment]
Hello @Franckapik !

_J'ai [simplifié](https://codesandbox.io/s/festive-williamson-grjsh?file=/src/3d/Joueur.js) votre exemple jusqu'à où je pense que nous pouvons voir le problème_: You were setting the player's position via the API, preventing collisions. I know this is not the file you directly linked to, but in looking through your game it seems you were not using `Game.js` at all.

In _`src/3d/Joueur.js`_, removing the following line will have your console light up with "hey"s. :)

```javascript
useFrame(() => api.position.set(y, 0.1, x));
```

**Comment by Franckapik** [comment]
Hello @stockHuman !

I'm sorry, i didn't realize that i updated my github so my codesandbox too. I was talking about Game.js file indeed where all meshes was imported from blender. Now, your simplified version (and thank you for this by the way !) is not using gltf... 
I will send a new version to explain better my situation. 
But, did you say that it's not possible to have a mesh controlled via api position (and so keyboard here) and have collision with it ? 

If i remove the line you said, the collision between "Joueur" and the ground works, but how to move the "Joueur" to play the game next ? 


Thank you very much for responding !

**Comment by stockhuman** [comment, marked as answer]
Unfortunately, I'm going to point you to the FAQ:

> A kinematic body moves under simulation according to its velocity. They do not respond to forces. They can be moved manually, but normally a kinematic body is moved by setting its velocity. A kinematic body behaves as if it has infinite mass. **Kinematic bodies do not collide with other static or kinematic bodies.**

Since your player, walls and floor are all Kinematic or static, you will not see collisions between them. 

Are you looking to perhaps `applyLocalImpulse` or force with every player movement? The player box would not move as you've defined it to so far, however. Or to rephrase that, are you looking for physics-based movement of the player, or logic based movement?

Regarding glTF, I'm not sure that's related to your problem. If you'll provide another sandbox with your progress we can see what's up.

**Comment by Franckapik** [comment]
Thank you @stockHuman . I will write roughly on a sheet of paper your citation about collision and kinematic body. So now, it's true that i have to make a choice between physic-based movement and logic ones... I prefered experiment the physics-based movement and i tried to use applyLocalImpulse but without sucess... I confess that i am a little embarrassed with the docs concerning this function. I didn't found example using it, have you got one ?

Thank you!!

**Comment by stockhuman** [comment]
Hello @Franckapik, pardon the late reply - I hope you managed to figure out what you want to go with for your game. Here's a very [basic demo](https://codesandbox.io/s/bouncing-diamond-mcr3b?file=/src/index.js) of `ApplyImpulse`, which you might want to use by perhaps applying an impulse from opposite the direction you want to move in, and _blasting_ your player in the desired direction. Good luck!

**Comment by Franckapik** [comment]
Thank you!! Now i see how to use ApplyImpulse. I have to decide if i prefer use Logic or Physic now to move my player ;)
